Source/Core/Core/PowerPC/JitArm64/JitArm64_LoadStoreFloating.cpp

```diff
@@ -164,9 +164,9 @@ void JitArm64::lfXX(UGeckoInstruction inst)
 
   BitSet32 regs_in_use = gpr.GetCallerSavedUsed();
   BitSet32 fprs_in_use = fpr.GetCallerSavedUsed();
-  regs_in_use[W0] = 0;
+  regs_in_use[static_cast<size_t>(ARM64Reg::W0)] = 0;
   fprs_in_use[0] = 0;  // Q0
-  fprs_in_use[VD - Q0] = 0;
+  fprs_in_use[static_cast<size_t>(VD - ARM64Reg::Q0)] = 0;
```

Comment:
Probably preferable:

```suggestion
  fprs_in_use[DecodeReg(VD)] = 0;
```

(Same for all similar lines, like `regs_in_use[DecodeReg(ARM64Reg::W0)] = 0;`)

Reply:
I can't follow this suggestion as given, because DecodeReg returns an ARM64Reg which (since it's now an enum class) doesn't implicitly convert to a size_t.  In order to use it as an index BitSet::operator[] I'd either have to cast it to a size_t anyway or write a function to perform that cast, or something else along those lines.

Given that DecodeReg masks its argument to [0,31] and most of the values in question are constants already in that range, the only purpose I can see to using DecodeReg in those cases would be for expressing intent.  Would you say it's worth adding in addition to a cast, or should I just stick with the cast?

As for replacing VD - Q0 with DecodeReg(VD), is VD guaranteed to be in the range [192, 223]?  Its value is set by fpr.RW(), whose behavior is rather opaque to me.

Reply:
Ah, somehow I thought it was returning an `int` :( Feel free to ignore my suggestion then (either by only using `DecodeReg` on lines that were doing this kind of subtraction in master, or by not using `DecodeReg` at all).

> As for replacing VD - Q0 with DecodeReg(VD), is VD guaranteed to be in the range [192, 223]? Its value is set by fpr.RW(), whose behavior is rather opaque to me.

Yes, in practice it always is. But in case it were not in that range (as you say, it isn't obvious that it is) then using `DecodeReg` would be necessary to get the correct behavior.